docs/sources/community/lids/_DRAFT_AsynchronousQueries.md

```diff
@@ -0,0 +1,105 @@
+
+* The query result without the ability to tune execution time or Loki's resource constraints.
+* The query result represents an expensive but important (aggregation) result set of logs and has a higher value as such irrespective how long it takes.
+
+## Goals
+
+The following proposals aim at providing the required infrastructure and query capabilities to execute queries in Loki asynchronously meaning the user will not fetch a result immediately but at later point in time using an identification for the requested query.
+
+## Non-Goals (optional)
+
+The following proposal do not aim at extending Loki to asynchronous capabilities beyond executing queries, e.g. offering asynchronous streams to read/write from/to Loki.
+
+## Proposals
+
+### Proposal 0: Do nothing
+
+The current state represents the do-nothing scenario very well, as per trade-offs currently to consider:
+
+* Users need to tune their queries manually (e.g. distinct label selectors, batching time frames).
+* Users need to request per-tenant and/or per-request limits by their respecting Loki administrators to overcome limits for certain query demand over time.
+* Client applications and networking infrastructure needs to be tuned for Loki response latency scenarios (e.g. Grafana datasource dataproxy, Server Write Timeouts on Gateways)
+
+### Proposal 1: Built-in asynchronous streaming component for asynchronous queries
+
+The main approach of a built-in asynchronous streaming component as devised in the [proof-of-concept PR](https://github.com/grafana/loki/pull/9730) is based on [nats.io's jetstream technology](https://nats.io/) and consists of:
+
+- A new Loki component called the `nats-server` that is hosting the persistence layer for any NATS JetStreams.
+- Multiple JetStream pub/sub streams for handling asynchronously Loki instant/range query requests and responses.
+
+The main benefits of this approach can be summarized as:
+
+- Provides a built-in stream handling component that can self-hosted with Loki as native Loki component (e.g. ring handling, discovery, scaling)
+- Allows full-control on stream characteristics (i.e. retention, message replication, ack policy)
+- Enables handling queries as simple pub/sub messages using built-in JetStream primitives (i.e. pub/sub provider as a native Loki subservice).
+
+The tradeoffs using this approach are:
+
+- A direct dependency to the NATS.io client and server code to be maintained.
+- A whole new set of settings to control and maintain JetStream streams on the server-side.
+- Object storage support for persisting streams currently experimental.
+
+### Proposal 2: Client support for external asynchronous streaming systems
```

Comment:
I'm less attracted to this option because it seems more complex in terms of external dependencies and thus less likely to see use.

Reply:
I feel similar about having just client-like implementations. However for the sake of completion and respecting the legacy of Loki (probably mostly based on how cortex evolved), I didn't want to leave this proposal out.